### ISSUE TITLE
Why to use null for data/error keys?

### ISSUE THREAD

**Issue by JerryGreen** [schettino/react-request-hook#11]
The topic about these two lines and some logic beneath them:

https://github.com/schettino/react-request-hook/blob/7a7da5cfa60da589cef342729bd55c3eba38b5d5/src/useResource.ts#L17-L18

It's already may be `undefined`. It should be sufficient to use `undefined`. Why should we make it `null` and/or place `null` there? (Do you remember the joke about the billion dollar mistake?)

**Comment by schettino** [comment]
Actually, `null` is the most accurate one, 'cause it's the default. The `undefined` is coming as a side effect though, as we tricky the compiler on the request helper to facilitate extracting the payload type later. We do that specially for apps not using typescript, so they can still have some type safety. I did remove the `undefined` with, apparently, no further changes. Will test a bit more and release a patch.

**Comment by JerryGreen** [comment]
@schettino I don't understand why to use null instead of undefined. What do you mean by "the default"? Where it comes from? It's defined by this library, right? Since there's no reason in both null and undefined I suggest to use undefined instead of null because this:

```
interface Args {
  token?: Token,
  router: RouterProps
}
```

Is better than this:

```
interface Args {
  token: Token | null,
  router: RouterProps
}
```

However, right now I have to use the worst:

```
interface Args {
  token?: Token | null,
  router: RouterProps
}
```

**Comment by schettino** [comment]
I think it makes sense having this form `data?: Token` but it may bring a breaking change, as some folks might be checking `data === null` somewhat. That's also the reason I haven't merged #12 yet. On the other hand, `error` might still become `null` (in the case where an error happens, and the following request succeeds). I have really exciting features to incorporate, but I won't be able to open a PR during the next week, but we can tackle all this in one major release.
